crates/sui-indexer/src/metrics.rs

```diff
@@ -95,6 +95,9 @@ pub struct IndexerMetrics {
     // indexer state metrics
     pub db_conn_pool_size: IntGauge,
     pub idle_db_conn: IntGauge,
+
+    pub address_processor_failure: IntCounter,
```

Comment:
in case processors get stuck, we will want to track the latest processed checkpoint on each processor so that we can monitor the delta between them and latest checkpoint on FN.